instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/Metric.java

```diff
@@ -46,13 +48,16 @@ public void setDesc(String desc) {
     this.desc = desc.trim();
   }
 
-  MetricInfo buildMetricInfo(
-      @Nullable String prefix,
-      String attributeName,
-      String defaultSourceUnit,
-      String defaultUnit,
-      MetricInfo.Type defaultType) {
+  /**
+   * @param attributeName attribute name
+   * @param jmxRule parent JMX rule where metric is defined
+   * @return metric info
+   * @throws IllegalStateException when effective metric definition is invalid
+   */
+  MetricInfo buildMetricInfo(String attributeName, JmxRule jmxRule) {
```

Comment:
[for reviewer] directly providing the parent JmxRule allows to make easier the "parent/child" inheritance where the fields value not explicitly defined at metric level are retrieved from the parent.

Reply:
nit: I know there is context in the doc block, but I think also making it more explicit that this is a parent rule by including it in the variable name would make the logic more clear

```suggestion
  MetricInfo buildMetricInfo(String attributeName, JmxRule parentJmxRule) {
```